include/xgboost/c_api.h

```diff
@@ -221,6 +155,189 @@ XGB_DLL int XGDMatrixCreateFromDT(void** data,
                                   bst_ulong ncol,
                                   DMatrixHandle* out,
                                   int nthread);
+
+/*
+ * ========================== Begin data callback APIs =========================
+ *
+ * Short notes for data callback
+ *
+ * There are 2 sets of data cakkbacks for DMatrix.  The first one is currently exclusively
```

Comment:
"cakkbacks" :)

This improved documentation is really nice.

Reply:
Oops.